.github/workflows/bump.yml

```diff
@@ -0,0 +1,187 @@
+name: Check and deploy API documentation
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  determine-doc-ids:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get changed files
+        id: changed-files
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            CHANGED_FILES=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }})
+          else
+            CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }})
+          fi
+          echo "CHANGED_FILES<<EOF" >> $GITHUB_ENV
+          echo "$CHANGED_FILES" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Set matrix
+        id: set-matrix
+        run: |
+          DOCS=()
+
+          # Check for changes in API doc folders
+          if echo "$CHANGED_FILES" | grep -q "^admin/"; then
+            DOCS+=("admin")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^cloud-controlplane/"; then
+            DOCS+=("cloud-controlplane")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^cloud-dataplane/"; then
+            DOCS+=("cloud-dataplane")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^schema-registry/"; then
+            DOCS+=("schema-registry")
+          fi
+          if echo "$CHANGED_FILES" | grep -q "^http-proxy/"; then
+            DOCS+=("http-proxy")
+          fi
+          # Check for changes in shared resources that might affect all docs
```

Comment:
Would it still be all right if the files in `shared/` don't necessarily affect all docs? Right now the files in that directory really only apply to the Cloud APIs.